### PR TITLE
Reverting the discovery job

### PR DESCRIPTION
**Pull request by dwertent**
## PR Type:
Refactoring

___
## PR Description:
This PR reverts the service discovery job and hard-codes the service discovery due to compatibility issues with ArgoCD caused by the addition of pre-install hooks. The discovery job has been removed and the discovery has been hard-coded instead. The chart and app versions have been updated to 1.15.6.

___
## PR Main Files Walkthrough:
<details> <summary>files:</summary>

`charts/kubescape-operator/Chart.yaml`: Updated the chart and app versions to 1.15.6.
`charts/kubescape-operator/templates/configs/cloudapi-configmap.yaml`: Removed pre-install and pre-upgrade hooks from the configmap and hard-coded the metrics and services.
`charts/kubescape-operator/templates/kollector/statefulset.yaml`: Removed pre-install and pre-upgrade hooks from the statefulset.
`charts/kubescape-operator/templates/kubescape/deployment.yaml`: Added a new environment variable 'KS_DEFAULT_CLOUD_CONFIGMAP_NAME' to the deployment.
`charts/kubescape-operator/templates/servicediscovery/job.yaml`: Disabled the service discovery job.
`charts/kubescape-operator/templates/servicediscovery/role.yaml`: Disabled the service discovery role.
`charts/kubescape-operator/templates/servicediscovery/rolebinding.yaml`: Disabled the service discovery rolebinding.
`charts/kubescape-operator/templates/servicediscovery/serviceaccount.yaml`: Disabled the service discovery serviceaccount.
`charts/kubescape-operator/values.yaml`: Renamed the cloudConfig from 'ks-cloud-config' to 'url-config' and added hard-coded values for metrics and services under serviceDiscovery.
</details>

___
## Overview
This PR reverts the discovery job and hard-codes the service discovery. The changes were made due to compatibility issues with ArgoCD caused by the addition of pre-install hooks. The discovery job has been removed and the discovery has been hard-coded instead.

### CODE REVIEW

**Review by matthyx — DISMISSED** [review]
PR checks fail because of https://github.com/helm-unittest/helm-unittest/pull/220

**Review by matthyx — DISMISSED** [review]

File: charts/kubescape-operator/values.yaml [hunk 1]
```diff
@@ -81,7 +81,7 @@ global:
   proxySecretName: kubescape-proxy-certificate
   namespaceTier: ks-control-plane
   capabilitiesConfig: ks-capabilities
-  cloudConfig: ks-cloud-config
+  cloudConfig: url-config
```
[the remark this answers is not in the document]
Comment (amirmalka) [reply charts/kubescape-operator/values.yaml:84]: But this is something we are aware of - if that's the case, we need to fix the code, and then update the chart

File: charts/kubescape-operator/values.yaml [hunk 2]
```diff
@@ -616,6 +616,9 @@ nodeAgent:
 serviceDiscovery:
   name: service-discovery
 
+  metrics: "otelcol.armosec.io:443"
+  services: '{"version":"v1","response":{"event-receiver-http":"https://report.armo.cloud","event-receiver-ws":"wss://report.armo.cloud","gateway":"wss://ens.euprod1.cyberarmorsoft.com","api-server":"https://api.armosec.io","metrics":"otelcol.armosec.io:443"}}'
```
Comment (amirmalka) [inline charts/kubescape-operator/values.yaml:620]: what about system tests?

File: charts/kubescape-operator/values.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (matthyx) [reply charts/kubescape-operator/values.yaml:84]: but I think in that case we set the env var, so the default value isn't used

**Comment by amirmalka** [comment]
I'm reading this: https://argo-cd.readthedocs.io/en/stable/user-guide/helm/#helm-hooks
What was exactly the issue with ArgoCD?

**Comment by matthyx** [comment]
> I'm reading this: https://argo-cd.readthedocs.io/en/stable/user-guide/helm/#helm-hooks What was exactly the issue with ArgoCD?

maybe https://github.com/argoproj/argo-cd/issues/15472

**Comment by amirmalka** [comment]
As discussed with @dwertent , I have removed the service discovery so that we don't have stale garbage in our chart.
If we need that logic we can take it from git for a better solution.

**Review by matthyx — APPROVED** [review]

**Comment by amirmalka** [comment]
We will not need this PR once this is merged: https://github.com/kubescape/helm-charts/pull/297

**Pull request closed without merging**